src/CMakeLists.txt

```diff
@@ -42,6 +42,7 @@ endif()
 if(UMF_BUILD_SHARED_LIBRARY)
     add_umf_library(unified_memory_framework SHARED ${UMF_SOURCES})
     target_compile_definitions(unified_memory_framework PUBLIC UMF_SHARED_LIBRARY)
+    target_link_libraries(unified_memory_framework numa)
```

Comment:
What if libbumf will have more dependencies?

I think instead of hardcoding particular dependencies in two places we need to link with `LIBS_UMF_STATIC`.
And `LIBS_UMF_STATIC` is a bad name in that case - I would rename it to `LIBS_UMF_DEPENDENCIES` or something like that.

So we should handle the single variable that capture UMF dependencies and

- if UMF is static lib link ubench directly with that dependencies
- if UMF is dynamic lib link UMF itself with the dependencies

Reply:
I still don't quite understand why we need to link those dependencies to tests. Do we use any libnuma APIs directly in tests? Otherwise, we should not need to link with it.

Reply:
See: https://github.com/oneapi-src/unified-memory-framework/pull/89

Reply:
Oh, good point. I did not know that static library can somehow contain information about its dependencies. This post explain that: https://stackoverflow.com/questions/43176681/do-statically-linked-libraries-require-linking-with-their-dependencies

Thank you, Igor!